### DISCUSSION TITLE
If we can move Lawnchair's sourceSets into their default dirs?

### DISCUSSION THREAD

**Discussion by Goooler** [LawnchairLauncher/lawnchair#3052] [Q&A] [answered]
### About this discussion

Cause I see the code highlight and navigation don't work with AS well, so if we can move the Lawnchair's sourceSets into their default dirs to support functions above. Link https://github.com/LawnchairLauncher/lawnchair/pull/2806#issuecomment-1210426545, we can just move Lawnchair own sourceSets, and let Lawnchair as a submodule instead of rootmodule.

**Comment by suphon-t** [comment]
Code highlighting and navigation have always been working since the start. There's probably an issue with your IDE (broken caches?). I doubt moving the source sets would fix anything.

**Comment by yasanglass** [comment]
It is not related to cache. This has been happening to me since 2-3 months ago. I have tried it on fresh installs of both Linux and windows installs and on 5+ devices (2 myself and 3 other people). Also, see #2647.

I did also talk about it with Patryk but he somehow didn't have the issue.

For me, it usually gets fixed on its own after a week or two of development but working on the project without any IDE features. The issue isn't just highlighted, you even have to manually import classes, etc.

What I have been doing recently to avoid the issue, is to reuse an old clone of the project since somehow it won't have this issue while new clones/projects do. I literally send a zip file to my coworkers to use instead of directly cloning the project.

@Goooler I can send you the file so you can work on the project without issues for now, but it's not a good enough solution.

**Comment by Goooler** [comment]
@yasandev Thanks for your explanation, I'll file an issue to the Android Studio team for help.

**Comment by yasanglass** [comment]
It is already reported

https://issuetracker.google.com/issues/232007221

**Comment by subashz** [comment]
> It is not related to cache. This has been happening to me since 2-3 months ago. I have tried it on fresh installs of both Linux and windows installs and on 5+ devices (2 myself and 3 other people). Also, see #2647.
> 
> I did also talk about it with Patryk but he somehow didn't have the issue.
> 
> For me, it usually gets fixed on its own after a week or two of development but working on the project without any IDE features. The issue isn't just highlighted, you even have to manually import classes, etc.
> 
> What I have been doing recently to avoid the issue, is to reuse an old clone of the project since somehow it won't have this issue while new clones/projects do. I literally send a zip file to my coworkers to use instead of directly cloning the project.
> 
> @Goooler I can send you the file so you can work on the project without issues for now, but it's not a good enough solution.

yeah I also had this issue. I usually do 'Repair IDE' option couple of time until that fix autocomplete and other stuff.

**Comment by Goooler** [comment]
Alright, just waiting for replies.

https://issuetracker.google.com/issues/232007221#comment31

**Comment by yasanglass** [comment]
Although, even if this issue is fixed, I would still prefer using the default dirs. Especially because it can make creating new flavors much easier.

**Comment by Goooler** [comment]
If other flavors are still necessary? I can't build them.

**Comment by Goooler** [comment, marked as answer]
Find a solution, seems we can try to "Mark directory as Sources Root" like this

<img width="1858" alt="image" src="https://user-images.githubusercontent.com/10363352/186302435-33deb684-9ecc-4261-bd89-698c3e3a0f3c.png">

**Comment by yasanglass** [comment]
> Find a solution, seems we can try to "Mark directory as Sources Root" like this

Haven't tried this solution but I no longer have this issue after running the project on Android Studio Electric Eel.
